app/controllers/user_confirmations_controller.rb

```diff
@@ -0,0 +1,37 @@
+class UserConfirmationsController < DeviseController
+  include Spree::Core::ControllerHelpers::Auth # Needed for access to current_ability, so we can authorize! actions
+
+  # GET /resource/confirmation/new
```

Comment:
here's [the controller in Devise](https://github.com/plataformatec/devise/blob/beeb48c404ae8b4c3c34f6d0a2325ca2bcc9298e/app/controllers/devise/confirmations_controller.rb), for comparison. Most notable change is the use of `authorize!`. To be honest I didn't know if Cancan would prevent from sending an email without that (that's what it's for, right?), I'll try without it to be sure. I also tried to use the original Devise controller (by having in the routes something like `:controllers => { :confirmations => 'devise/confirmations' }` but I couldn't make it work.